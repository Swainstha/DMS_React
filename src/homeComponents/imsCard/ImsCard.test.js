import React from 'react';
import ImsCard from './ImsCard';
import {
    configure,
    shallow
} from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
// JestHook.mock('expo-font');
configure({
    adapter: new EnzymeAdapter
})


const setup = (props = {}, state = null) => {
    return (shallow( < ImsCard {...props}  />)
    )
}

const findByTestAttr=(wrapper, val)=>{
    return wrapper.find(`[data-test='${val}']`)
}


const dummy_data=
    {name:"8",
    type:{
        image:"image",
        name:"name",
        type:"type1"
    }}


describe('ImsCard />', () => {
  let wrapper;


  beforeEach(() => {
    wrapper = setup(dummy_data)
  });

  afterEach(() => {
    jest.clearAllMocks();
  });


it("ImsCard snapshot", () => {
        // let wrapper = setup(dummy_data);
        expect(wrapper).toMatchSnapshot();

  
})

it("ImsCard Components", () => {
    const linkComponent= findByTestAttr(wrapper,"link-component")
    const img = wrapper.find('img')
    const h3 = wrapper.find('h3')
    expect(linkComponent.prop('to').aboutProps).toEqual({type:"type1"})
    expect(img.prop("src")).toEqual("image")
    expect(img.prop("alt")).toEqual("name")
    expect(h3.text()).toEqual("name")


})
})
