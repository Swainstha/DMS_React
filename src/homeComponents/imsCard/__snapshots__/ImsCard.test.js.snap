// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImsCard /> ImsCard snapshot 1`] = `ShallowWrapper {}`;
